extension_service/datastore/providers/cloudsql.py

```diff
@@ -0,0 +1,464 @@
+                        ),
+                        parameters={
+                            "id": id,
+                        },
+                    )
+                )
+                .mappings()
+                .fetchone()
+            )
+
+        if result is None:
+            return None
+
+        res = models.Airport.model_validate(result)
+        return res
+
+    async def get_airport_by_iata(self, iata: str) -> Optional[models.Airport]:
+        async with self.__pool.connect() as conn:
+            result = (
+                (
+                    await conn.execute(
+                        sqlalchemy.text(
+                            """
+                      SELECT * FROM airports WHERE iata ILIKE :iata
+                    """
+                        ),
+                        parameters={
+                            "iata": iata,
+                        },
+                    )
+                )
+                .mappings()
+                .fetchone()
+            )
+
+        if result is None:
+            return None
+
+        res = models.Airport.model_validate(result)
+        return res
+
+    async def search_airports(
+        self,
+        country: Optional[str] = None,
+        city: Optional[str] = None,
+        name: Optional[str] = None,
+    ) -> list[models.Airport]:
+        async with self.__pool.connect() as conn:
+            results = (
+                (
+                    await conn.execute(
+                        sqlalchemy.text(
+                            """
+                    SELECT * FROM airports
+                    WHERE (CAST(:country AS TEXT) IS NULL OR country ILIKE :country)
+                    AND (CAST(:city AS TEXT) IS NULL OR city ILIKE :city)
+                    AND (CAST(:name AS TEXT) IS NULL OR name ILIKE '%' || :name || '%')
+                    """
+                        ),
+                        parameters={
+                            "country": country,
+                            "city": city,
+                            "name": name,
+                        },
+                    )
+                )
+                .mappings()
+                .fetchall()
+            )
+
+        res = [models.Airport.model_validate(r) for r in results]
+        return res
+
+    async def get_amenity(self, id: int) -> Optional[models.Amenity]:
+        async with self.__pool.connect() as conn:
+            result = (
+                (
+                    await conn.execute(
+                        sqlalchemy.text(
+                            """
+                    SELECT id, name, description, location, terminal, category, hour
+                    FROM amenities WHERE id=:id
+                    """
+                        ),
+                        parameters={
+                            "id": id,
+                        },
+                    )
+                )
+                .mappings()
+                .fetchone()
+            )
```

Comment:
don't be afraid to not nest, especially on complicated/long statements like this -- it often improves readability to pull it out.
```suggestion
            s = text(
                """
                SELECT id, name, description, location, terminal, category, hour
                FROM amenities WHERE id=:id
                """
            )
            p = {  "id": id }
            result = await conn.execute(s, parameters=p).mappings().fetchone()
```